rust/src/mqtt/logger.rs

```diff
@@ -73,7 +75,15 @@ fn log_mqtt(tx: &MQTTTransaction, flags: u32, js: &mut JsonBuilder) -> Result<()
                         js.set_string("topic", will_topic)?;
                     }
                     if let Some(will_message) = &conn.will_message {
-                        js.set_string_from_bytes("message", will_message)?;
+                        if max_log_len > 0 {
```

Comment:
I would rather not log it at all, than log it without limits

Reply:
I can imagine that in the absolute majority of MQTT use cases, this will be a small text value, and probably also the most interesting value in the protocol.  The protocol allows for at most 256MB as the length of one message, which is bounded by default in the parser to 1MB.

If we decide to _not_ log any messages unless a limit is set, and the current default config value of 0 is used, this value would just be missing in the log (which would likely not be what the user wants). So we indeed need to decide on a default for that -- do we want to pick some arbitrary limit? Since we typically expect a small string, should we maybe just use 1KB as the default? Any ideas/comments/objections?

Reply:
After changing the default limit to 1kb, one would actively need to set the limit to 0 in the configuration to explicitly enable unbounded logging (e.g. for debugging outside of production). I think this is better than just outright disabling unbounded logging, and definitely better than using no limit as the default.